docs/src/index.md

```diff
@@ -134,6 +134,29 @@ Test Summary:                    | Pass  Total
 relu at -0.5, with cotangent 1.0 |    4      4
 ```
 
+## Custom finite differencing
+
+If a package is using a custom finite differencing method of testing the `frule`s and
+`rrule`s, `check_equal` function provides a convenient way of comparing [various types]
+(https://www.juliadiff.org/ChainRulesCore.jl/dev/design/many_differentials.html#Design-Notes:-The-many-to-many-relationship-between-differential-types-and-primal-types.)
+of differentials.
+
+It is effectively `isapprox` but with better error messages because it knows about `thunk`s
+and `ChainRules` differential types `Zero()`, `DoesNotExist()`, and `Composite`.
+It is not simply overloading `isapprox` because then comparing `Tuple`s/`NamedTuple`s
```

Comment:
It is basically `(a,b) -> @test isapprox(a, b)`
so it is different to overloading `isapprox` since it doesn't return true or false, it adds a failure or a pass to the current testset.

The reason it isn't overloading `isapprox` is because sometimes we need to really fiddle with the inputs before putting them into the `@test`
and `@test isapprox(a,b)` will show `a` and `b`'s value on a failure.
But actually we want to fiddle anound with `a` and `b` to give them more readable values before checking them.
Such as `unthunk`ing.

Reply:
Could this not be achieved by something like:
```
function Base.ispprox(x::AbstractThunk, y)
    return isapprox(unthunk(x), y)
end
```
It feels that adding `@test` inside is a matter of style?

Reply:
Nope.

because `@test` only looks 1 level deep when displaying output.

```julia
julia> struct Foo end

julia> unfoo(::Foo) = 1
unfoo (generic function with 1 method)

julia> unfoo(x) = x
unfoo (generic function with 2 methods)

julia> Base.isapprox(foo::Foo, b; kwargs...) = isapprox(unfoo(foo), b)

julia> using Test

julia> @test isapprox(Foo(), 2)
Test Failed at REPL[10]:1
  Expression: isapprox(Foo(), 2)
   Evaluated: isapprox(Foo(), 2)
ERROR: There was an error during testing
```

Notice that it displayed `isapprox(Foo(), 2)` for the `Evaluated` line.

What we want to be displayed is more like:
```julia

julia> a = 1
1

julia> @test isapprox(a, 2)
Test Failed at REPL[12]:1
  Expression: isapprox(a, 2)
   Evaluated: isapprox(1, 2)
ERROR: There was an error during testing
````

Reply:
Thanks, all clear now!